src/pages/trainer/training/edit/components/editor.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import {ToolbarContainerComponent, IToolbarCommand} from './toolbar';
+import {textAreaTool} from '../../../../../common/ui/tools/textAreaTool';
+const classNames: any = require('./editorStyles.scss');
+
+interface IProps {
+  content: string;
+  cursorStartPosition: number;
+  shouldUpdateEditorCursor: boolean;
+  toolbarCommand: IToolbarCommand;
+  className: string;
+  onContentChange: (content: string) => void;
+  updateEditorCursor: (cursorStartPosition: number) => void;
+}
+
+export class EditorComponent extends React.Component<IProps, {}> {
+  private editor: HTMLTextAreaElement;
+  private refHandlers = {
+    textArea: (textArea) => { this.editor = textArea; },
+  };
+
+  public componentWillReceiveProps(nextProps: IProps) {
+    if (nextProps.toolbarCommand !== this.props.toolbarCommand) {
+      this.updateContentWithToolbarCommand(nextProps.toolbarCommand);
+      this.updateEditorCursor(nextProps.toolbarCommand.offset);
+    }
+  }
+
+  private updateContentWithToolbarCommand(toolbarCommand: IToolbarCommand) {
+    const editorContent = textAreaTool.insertAtCaretGetText(this.editor, toolbarCommand.caret, toolbarCommand.offset);
+    this.props.onContentChange(editorContent);
+  }
+
+  private updateEditorCursor(offset: number) {
+    const cursorStartPosition = textAreaTool.calculateStartCursorPositionPlusOffset(this.editor, offset);
+    this.props.updateEditorCursor(cursorStartPosition);
+  }
+
+  public componentDidUpdate() {
+    if (this.props.shouldUpdateEditorCursor) {
```

Comment:
Just thinking about edge cases, e.g. content doesn't change but you move with the cursor and the just add some command to insert a bold or whatever.